### PR TITLE
Configure queues from workers instead of yml file

### PR DESCRIPTION
**Pull request by fzf**

### CODE REVIEW

File: lib/simplekiq.rb [hunk 1]
```diff
@@ -1,14 +1,23 @@
...
 require 'simplekiq/worker'
+require 'string'
 
 module Simplekiq
   class Error < StandardError; end
   # Your code goes here...
-  def self.start
-    Datadog.config
+  def self.config
+    Datadog.new.config
```
Comment (sidt101) [inline lib/simplekiq.rb]: [non-blocker] can the `config` methods be class methods?

File: lib/simplekiq/queue_getter.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      '-' unless app_name.nil?
+    end
+
+    def app_name
+      Simplekiq.app_name
+    end
+
+    def worker_classes
+      ObjectSpace.each_object(Class).select do |klass|
+        # binding.pry if klass.included_modules.include? Simplekiq::Worker
```
Comment (sidt101) [inline lib/simplekiq/queue_getter.rb]: We should remove this comment

File: spec/lib/simplekiq/config_spec.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
+require 'spec_helper'
+
+RSpec.describe Simplekiq::Config do
+  describe '#config' do
+    before do
+      allow_any_instance_of(
```
Comment (sidt101) [inline spec/lib/simplekiq/config_spec.rb]: [non-blocker] I'd like us to move away from using `any_instance_of` for reasons listed here https://relishapp.com/rspec/rspec-mocks/docs/working-with-legacy-code/any-instance

File: lib/simplekiq/datadog.rb [hunk 4]
```diff
@@ -3,20 +3,24 @@
...
-                "service:#{Rails.application.class.parent_name.downcase rescue 'undefined'}"
+                "service:#{app_name}"
               }
             ]
            )
         end
       end
     end
+
+    def app_name
```
Comment (sidt101) [inline lib/simplekiq/datadog.rb]: [non-blocker] We can also think about moving this definition of `app_name` to simplekiq.rb and redefining this method in all other classes.

Comment (fzf) [reply lib/simplekiq/datadog.rb]: Leaving this open in other classes in case override is necessary

**Pull request merged**